sound/soc/sof/ipc4-topology.c

```diff
@@ -889,6 +913,15 @@ static int sof_ipc4_widget_setup_comp_process(struct snd_sof_widget *swidget)
 		goto err;
 	}
 
+	ret = sof_update_ipc_object(scomp, process->pin_info,
+				    SOF_PIN_FORMAT_TOKENS, swidget->tuples,
+				    swidget->num_tuples, sizeof(struct sof_ipc4_pin_info),
+				    swidget->num_sink_pins + swidget->num_source_pins);
```

Comment:
Do we require that _all_ pins must have format specified? If the module have only single src and sink pins then by definition the formats are derived from the common format and there will be no pin format in tplg.

Reply:
@ujfalusi requiring the parsing of swidget->num_sink_pins + swidget->num_source_pins doesnt mean that we require the tplg to have set all the pin formats. This function will try to find a max of that many pin formats but will not complain if there are less than that. So if a module doesnt use all its pins, the tplg need only have as many pin formats as needed